components/tabs/src/storage.rs

```diff
@@ -77,20 +157,65 @@ impl TabsStorage {
         None
     }
 
-    pub fn get_remote_tabs(&self) -> Option<Vec<ClientRemoteTabs>> {
-        self.remote_tabs.borrow().clone()
+    pub fn get_remote_tabs(&mut self) -> Option<Vec<ClientRemoteTabs>> {
+        match self.open_if_exists() {
+            Err(e) => {
+                log::error!("Failed to read remote tabs: {}", e);
+                None
+            }
+            Ok(None) => None,
+            Ok(Some(c)) => {
+                match c.query_rows_and_then_named_cached(
+                    "SELECT payload FROM tabs",
+                    &[],
+                    |row| -> Result<_> { Ok(serde_json::from_str(&row.get::<_, String>(0)?)?) },
+                ) {
+                    Ok(crts) => {
+                        // only return clients that actually have tabs.
```

Comment:
Best I can tell, the old code doesn't filter out clients with no open tabs - presumably it relies on the consumer doing this, and there might even be a use-case for knowing the clients exist even if they currently have no tabs? So I'm removing this filtering.